skimage/feature/corner.py

```diff
@@ -67,41 +67,59 @@ def structure_tensor(image, sigma=1, mode='constant', cval=0):
     cval : float, optional
         Used in conjunction with mode 'constant', the value outside
         the image boundaries.
+    order : {'rc', 'xy'}, optional
+        This parameter allows for the use of reverse or forward order of
+        the image axes in gradient computation. 'rc' indicates the use of
+        the first axis initially (Arr, Arc, Acc), whilst 'xy' indicates the
+        usage of the last axis initially (Axx, Axy, Ayy).
 
     Returns
     -------
-    Axx : ndarray
+    Arr : ndarray
         Element of the structure tensor for each pixel in the input image.
-    Axy : ndarray
+    Arc : ndarray
         Element of the structure tensor for each pixel in the input image.
-    Ayy : ndarray
+    Acc : ndarray
         Element of the structure tensor for each pixel in the input image.
 
     Examples
     --------
     >>> from skimage.feature import structure_tensor
     >>> square = np.zeros((5, 5))
     >>> square[2, 2] = 1
-    >>> Axx, Axy, Ayy = structure_tensor(square, sigma=0.1)
-    >>> Axx
+    >>> Arr, Arc, Acc = structure_tensor(square, sigma=0.1, order='rc')
+    >>> Acc
     array([[0., 0., 0., 0., 0.],
            [0., 1., 0., 1., 0.],
            [0., 4., 0., 4., 0.],
            [0., 1., 0., 1., 0.],
            [0., 0., 0., 0., 0.]])
 
     """
+    if order is None:
+        if image.ndim == 2:
+            # The legacy 2D code followed (x, y) convention, so we swap the
+            # axis order to maintain compatibility with old code
+            warn('deprecation warning: the default order of the structure '
+                 'tensor values will be "row-column" instead of "xy" starting '
+                 'in skimage version 0.20. Use order="rc" or order="xy" to '
+                 'set this explicitly.', category=FutureWarning, stacklevel=2)
+            order = 'xy'
+        else:
+            order = 'rc'
 
     image = _prepare_grayscale_input_2D(image)
 
-    imx, imy = _compute_derivatives(image, mode=mode, cval=cval)
+    derivatives = _compute_derivatives(image, mode=mode, cval=cval)
+
+    if order == 'rc':
+        derivatives = reversed(derivatives)
```

Comment:
I'm not 100% sure if this is correct in 3D/nD. We can worry about that in the 3D/nD PR, though.

For example, mode `'xy'` (vs. 'yz') in NumPy's meshgrid only swaps the order of the first two while the rest remain the same. See the [Notes section here](https://numpy.org/doc/stable/reference/generated/numpy.meshgrid.html). This would instead reverse **all** axes for "xy" vs "rc". That may be correct, but I don't recall offhand what is done in other functions in `skimage`.

Reply:
This type of potential confusion is why I am keen to see the switch to rc convention as that seems the easiest to follow

Reply:
@grlee77 I suggest that we do not support 'xy' for ndim>2. It is only a temporary crutch until everything is rc.

Reply:
>  I suggest that we do not support 'xy' for ndim>2

Okay, then perhaps we should raise an error for order = 'xy' when ndim != 2

Reply:
Awesome! That will make the next step much easier. How do we handle this exactly? If we get an image with ndim>2 and order='xy' should we raise an error or warning or just silently ignore the option?

Reply:
I would raise an error